git-ext/src/reference/name.rs

```diff
@@ -76,15 +75,15 @@ impl RefLike {
     /// If `base` is not a prefix of `self`, or `base` equals the path in `self`
     /// (ie. the result would be the empty path, which is not a valid
     /// [`RefLike`]).
-    pub fn strip_prefix<P: AsRef<Path>>(&self, base: P) -> Result<Self, StripPrefixError> {
+    pub fn strip_prefix<P: AsRef<str>>(&self, base: P) -> Result<Self, StripPrefixError> {
```

Comment:
I'm a bit uncomfortable with having to remember to add a trailing slash to `base`. How about something like this?:

```rust
    let suffix = iter::repeat("").take(self.0.len().saturating_sub(base.len()));

    let path = self.0
        .split('/')
        .zip(base.split('/').chain(suffix))
        .skip_while(|(a, b)| a == b)
        .map(|(x, _)| x)
        .collect::<Vec<&str>>()
        .join("/");
```

Reply:
I agree with the concern on forgetting the trailing slash, but I was also concerned of doing too much work as well to fix it.

What about:

```rust
let base = format!("{}/", base.strip_suffix('/').unwrap_or(base))
```

Reply:
Mmya, although this is not technically doing _less work_...

I'm wary about the allocations introduced by `format!` everywhere -- I suspect this will be noticable eventually, as we're creating a lot of refnames all the time. But I'd be willing to accept this for now, and spend some time optimising in a separate patch.